project/BillyNI-OUO/README.rst

```diff
@@ -0,0 +1,64 @@
+==============================================================================
+MinSteinerTree: A Comparative Analysis of Minimum Steiner Tree and FLUTE Algorithms
+==============================================================================
+
+A project that implement a Minimum Steiner Tree Construction and compare to FLUTE Algorithms.
+
+Basic Information
+=================
+
+In the field of VLSI, the challenge of global routing has become increasingly important, and finding a good Minimum Steiner Tree algorithm is crucial. Without an efficient algorithm, the time complexity can increase and global routing quality can suffer. By improving the Minimum Steiner Tree algorithm, we can reduce the time complexity and improve the overall performance of the VLSI system. With this in mind, it is essential to develop and compare effective algorithms for constructing Minimum Steiner Trees.
+
+Problem to Solve
+================
+
+Global routing is a critical step in the VLSI design process. To simplify the problem, we model the design as a grid graph, taking into account cells, pins, and tracks. The key challenge in this process is finding a minimal Steiner tree that can connect all the pins and nets. This task is essential for achieving an optimal layout and reducing the overall complexity of the VLSI system.
+
+Prospective Users
+=================
+
+VLSI design engineers, researchers, and semiconductor manufacturers looking for an efficient and adaptive routing solution for their designs.
+
+System Architecture
+===================
+
+This software will be developed using C++ for computational work and Python with Pybind for the interface. The system will consist of the following components(layer).
+
+1. A Python IO interface for users to interact with program.
+2. A Python logic layer to handle useing right algorithm at the lower layer.
+3. Algorithm implement by C++. 
+
+
+API Description
+===============
+
+* createGraph(x, y): create a routing space with x * y grid.
+* addNetlist(netlist): add a netlist containing the nets, chip size, wire cost, and other related information.

```

Comment:
it's a data structure describe a net, in the simplest version it just contain mutiple pin coordinate.
Further version may cosider for mutiple net, it will have a list of net.

Reply:
A plan describing how the data structure looks like will speed up your implementation.